File: Testes automatizados/Selenium em Python com BDD/pages/cart_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import Select

class CartPage:
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(self.driver, 10)
        self.BOTAO_REMOVER = (By.CSS_SELECTOR, '[data-test="remove-sauce-labs-backpack"]')
        self.ITEM_CARRINHO = (By.CSS_SELECTOR, '[data-test="inventory-item-name"]')


    def clicarRemover(self):
        self.driver.find_element(*self.BOTAO_REMOVER).click()

    def verificarItem(self):
        elementos = self.driver.find_elements(*self.ITEM_CARRINHO)
        return len(elementos) > 0

    def verificarDoisItens(self):
        elementos = self.driver.find_elements(*self.ITEM_CARRINHO)
        return len(elementos) > 1


    

File: Testes automatizados/Selenium/Selenium/src/test/java/pages/CartPage.java
package pages;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;


public class CartPage {
    private WebDriver driver;

    private By removerItem = By.cssSelector("[data-test='remove-sauce-labs-backpack']");
    private By itemNoCarrinho = By.cssSelector("[data-test='inventory-item-name']");
    private By nomeItem = By.cssSelector("[data-test='inventory-item-name']");

    public CartPage(WebDriver driver) {
        this.driver = driver;
    }

    public void clicarRemoverItem()
    {
        driver.findElement(removerItem).click();
    }

    public boolean itemContemTexto(String textoEsperado) {
        return driver.findElements(nomeItem)
                .stream()
                .anyMatch(elemento -> elemento.getText().contains(textoEsperado));
    }

}



